app/patient-details/controllers/patientDetailController.js

```diff
@@ -21,4 +21,54 @@ angular.module('patient.details')
             $scope.linkDashboard = function() {
                 $location.url("/dashboard/" + patientUuid); // path not hash
             };
+                    
+            // print the demographics info
+            $scope.print = function(){
+                
+                var templateUrl = "../patient-details/views/patient-demo-info.html";
+                var data = $scope.patient;
+                
+                //you need to transform this piece of code in a service
+                var printHtml = function (html) {
+                    var deferred = $q.defer();
+                    var hiddenFrame = $('<iframe style="display: none"></iframe>').appendTo('body')[0];
+
+                    hiddenFrame.contentWindow.printAndRemove = function() {
+                        hiddenFrame.contentWindow.print();
+                        $(hiddenFrame).remove();
+                    };
+
+                    var htmlContent = "<!doctype html>"+
```

Comment:
Hi @steliomo, can we separate the template to be an external file? This way we can reuse it. Actually I think it would be better if we are able to reuse this entire printing mechanism.